core-tests/shared/src/test/scala/zio/PromiseSpec.scala

```diff
@@ -29,10 +44,25 @@ object PromiseSpec
         },
         testM("fail a promise using complete") {
           for {
-            p <- Promise.make[String, Int]
-            s <- p.complete(IO.fail("error with done"))
-            v <- p.await.run
-          } yield assert(s, isTrue) && assert(v, fails(equalTo("error with done")))
+            p  <- Promise.make[String, Int]
+            r  <- Ref.make[List[String]](List("first error", "second error"))
```

Comment:
Cosmetic: do we need type ascription here?

Reply:
Great point! We definitely don't need that, Fixed now.